src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -27,6 +27,9 @@ class SchemaCompareService
         private Lazy<ConcurrentDictionary<string, SchemaComparisonResult>> schemaCompareResults =
             new Lazy<ConcurrentDictionary<string, SchemaComparisonResult>>(() => new ConcurrentDictionary<string, SchemaComparisonResult>());
 
+        // For testability
+        internal Task CurrentSchemaCompareTask;
```

Comment:
what is this used for?

Reply:
for waiting in test to ensure the completion

Reply:
This doesn't cause any problems if  there are multiple schema compares sending requests right?

Reply:
No it wouldn't.. it will hold only the latest one  (It's only assigned not used anywhere in real code. For ref - query editor follows same pattern for test)